processes-endpoint.md

```diff
@@ -0,0 +1,215 @@
+* `status`: The status of the script
+* `parameter.xyz`: Which parameters have been used, and their value
+
+Sort options should support:
+* `script`
+* `startTime`
+
+## Execution Details
+**GET /api/system/processes/<:process-id>**
+
+This endpoint will return details on the requested process.
+
+```json
+{
+  "processId" : "000003f1-a850-49de-af03-997272d834c9",
+  "userId" : "aa0263e2-b90a-4528-89fa-116ea4859de1",
+  "startTime" : "2017-11-22T10:29:11Z",
+  "status" : "RUNNING",
+  "parameters" : [
+    {
+    "name" : "c",
+    "value" : "954e5cfa-6990-4c85-ae42-f30d8c7888e2"
+    },
+    {
+    "name" : "n",
+    "value" : "true"
+    }
+  ],
+  "_links" : {
+    "self" : {
+      "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9"
+    },
+    "script" : {
+      "href" : "/api/system/scripts/import"
+    },
+    "output" : {
+      "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9/output"
+    },
+    "files" : {
+      "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9/files"
+    }
+  }
+}
+```
+
+The possible `status` values are `RUNNING`, `COMPLETED` and `FAILED`.
+
+## Execution Console Output
+**GET /api/system/processes/<:process-id>/output**
+
+This endpoint will return the console output of the script. To keep the scope of this work limited, it will print the full console output without datestamps.
+In a future version, a solution supporting a live-feed can also be included.
+
+## Execution File Output List
+**GET /api/system/processes/<:process-id>/files**
+
+This endpoint will let an administrator download an output file created by a process. If the file is found, it will be presented as a download. This endpoint will support "Range" HTTP headers so that downloads can be paused and resumed.
+
+```json
+{
+  "processId" : "000003f1-a850-49de-af03-997272d834c9",
+  "_links" : {
+    "self" : {
+      "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9/files"
+    },
+    "mapfile" : {
+      "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9/files/mapfile"
+    },
+    "zipfile" : {
+      "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9/files/zipfile"
+    }
+  }
+}
+```
+
+## Execution File Output
+**GET /api/system/processes/<:process-id>/files/<:file-name>**
+
+This endpoint will let an administrator download an output file created by a process. If the file is found, it will be presented as a download. This endpoint will support "Range" HTTP headers so that downloads can be paused and resumed.
+
+Required parameters:
+* `file-name`: The name of the output file to retrieve
+
+## Execution Deletion
+**DELETE /api/system/processes/<:process-id>**
+
+The will delete the process and associated files and output.
+
+Support for stopping a process can be included in a future version
+
+### Script Invocation
+**POST /api/system/scripts/<:script-name>/processes**
```

Comment:
Shouldn't this be in the `scripts-endpoint.md` file?

Reply:
Agreed with @AlexanderS here.  This seems out of place, as it's a `/scripts` endpoint request listed in the `/processes` endpoint documentation. This should move to `scripts-endpoint.md`, though it is reasonable to add a reference in this file saying that `POST` isn't supported, and please use `/api/system/scripts/<:script-name>/processes` instead.

Reply:
I've moved the section now